src/libertem/io/dataset/base/backend.py

```diff
@@ -131,8 +138,16 @@ def _get_tiles_straight(self, tiling_scheme, fileset, read_ranges):
                 origin=origin,
                 shape=Shape(shape, sig_dims=sig_dims)
             )
-            data_slice = tile_slice.get()
-            data = memmap[data_slice]
+            if sync_offset == 0:
+                data = memmap[tile_slice.get()]
+            elif sync_offset > 0:
+                sig_origin = origin[-sig_dims:]
+                origin = (origin[0] + sync_offset,) + tuple(sig_origin)
+                tile_slice_with_offset = Slice(
+                    origin=origin,
+                    shape=Shape(shape, sig_dims=sig_dims)
```

Comment:
`shape` can be `tile_slice.shape` here, right? We don't necessarily have to construct a new `Shape` object, as only the `origin` is different.